### ISSUE TITLE
Cannot access member "clicked" for type "QPushButton";   Member "clicked" is unknown

### ISSUE THREAD

**Issue by Pickaxe233** [microsoft/pylance-release#3688]
# Environment data
* Language Server version: 2022.11.32
* OS and version: Windows 11 22H2
* Python version (& distribution if applicable, e.g. Anaconda): 3.10.8
# Expected behaviour
extension sets correct syntax highlighting
# Actual behaviour
extension can't find QPushButton class's clicked method
# Code Snippet / Additional information
```
class MainWindow(QMainWindow, Ui_MainWindow):
    def __init__(self, parent=None):
        super(MainWindow, self).__init__(parent)
        self.setupUi(self)

        self.pushButton.clicked.connect(self.search)
```
It's kind of like #884, and if I run the full script, everything is working as expected, but extension sets wrong syntax highlighting, and I'm using PySide6

**Comment by bschnurr** [comment]
Most likely and issue with missing stub info for 'clicked'

**Comment by Pickaxe233** [comment]
So could you please tell me how to solve it or am I able to solve it?

**Comment by Pickaxe233** [comment]
> Most likely and issue with missing stub info for 'clicked'

So could you please tell me how to solve it or am I able to solve it?

**Comment by erictraut** [comment]
@Pickaxe233, could you please provide a minimal, self-contained code snippet? Your code sample references symbols that are presumably imported from a library or are defined elsewhere in your code. If imported, include the import statements. And if they are coming from a library, please specify the version of the library and whether you have any type stubs installed.

**Comment by Pickaxe233** [comment]
> @Pickaxe233, could you please provide a minimal, self-contained code snippet? Your code sample references symbols that are presumably imported from a library or are defined elsewhere in your code. If imported, include the import statements. And if they are coming from a library, please specify the version of the library and whether you have any type stubs installed.

the bolded codes in the following codes are wrongly highlighted codes
<pre>
self.pushButton.<b>clicked</b>.connect(self.search)
self.pushButton_2.<b>clicked</b>.connect(self.voice1)
self.pushButton_3.<b>clicked</b>.connect(self.voice2)
self.pushButton_4.<b>clicked</b>.connect(lambda:self.pic(0))
self.pushButton_5.<b>clicked</b>.connect(lambda:self.pic(1))
self.actionConfigrations.<b>triggered</b>.connect(self.configs)
self.lineEdit.<b>returnPressed</b>.connect(self.search)
self.listWidget.<b>customContextMenuRequested</b>.connect(self.copy)
self.listWidget_2.<b>customContextMenuRequested</b>.connect(self.copy1)
self.textBrowser.<b>customContextMenuRequested</b>.connect(self.copy2)
self.checkBox.<b>stateChanged</b>.connect(lambda:self.pic(2))
self.pushButton_6.<b>clicked</b>.connect(self.translate)
self.pushButton_7.<b>clicked</b>.connect(self.clear)
</pre>
Here's all imports
```python
import json
import requests
import datetime
import sys
import re
import cn2an
import Threads
import config
from borax.calendars.lunardate import LunarDate
from PySide6.QtWidgets import QApplication, QMainWindow, QMessageBox, QMenu
from PySide6.QtGui import QImage, QPixmap, QFont, QFontDatabase
from ui_dict import Ui_MainWindow
from PySide6.QtMultimedia import QMediaPlayer, QAudioOutput, QMediaDevices, QAudioDevice
from PySide6.QtCore import Qt, QSize, QUrl
```
And all methods or others I have referenced haven't wrong highlights, for example
```python
 def clear(self):
    self.plainTextEdit.clear()
    self.textBrowser.clear()
```
except for methods include `Threads.voiceThread(t=1)`, and `Threads.voiceThread(t=1)` itself is highlighted wrongly, like:
<img width="395" alt="Screenshot_20221201_091249" src="https://user-images.githubusercontent.com/72426723/204942284-f72cb2a1-3661-4b27-bedd-07a226498997.png">
It's from Threads.py, but there aren't highlights in the whole file.
```python
from PySide6.QtCore import QThread, Signal

class voiceThread(QThread):
    finishSignal = Signal(int)

    def __init__(self, t, parent=None):
        super(voiceThread, self).__init__(parent)
        self.t = t

    def run(self):
        self.finishSignal.emit(int(self.t)) 
```

**Comment by bschnurr** [comment]
Thank you for the additional info.

`Threads.voiceThread(t=0)` seems to work for me after `import Threads`

if you `go to definition` on `Threads` in the statement `Threads.voiceThread` where does it take you?

**Comment by Pickaxe233** [comment]
> Thank you for the additional info.
> 
> `Threads.voiceThread(t=0)` seems to work for me after `import Threads`
> 
> if you `go to definition` on `Threads` in the statement `Threads.voiceThread` where does it take you?

It takes me to the file `Threads.py`
![example](https://user-images.githubusercontent.com/72426723/209133179-eea6b171-38d5-4bca-a7db-30e29cfcd7a0.gif)

**Comment by bschnurr** [comment]
can you paste the actual error text.. and also show me how you r importing "Threads"? ie. `import Threads`

**Comment by Pickaxe233** [comment]
> can you paste the actual error text.. and also show me how you r importing "Threads"? ie. `import Threads`

The actual error text:
> Cannot access member "clicked" for type "QPushButton"
  Member "clicked" is unknown
![image](https://user-images.githubusercontent.com/72426723/217181714-61a838ec-55d7-4b06-bc94-fadbfce3b9b4.png)

And I just use `import Threads` for importing

**Comment by erictraut** [comment]
@Pickaxe233, could you please provide a minimal, self-contained code snippet that demonstrates the problem you're seeing? Screen shots and incomplete code samples are not very useful in diagnosing issues like this.

**Comment by Pickaxe233** [comment]
> @Pickaxe233, could you please provide a minimal, self-contained code snippet that demonstrates the problem you're seeing? Screen shots and incomplete code samples are not very useful in diagnosing issues like this.

In `main.py`:
```python
from ui_main import Ui_MainWindow
#ui_main.py is main.ui's coverted version
from PySide6.QtWidgets import QApplication, QMainWindow
from PySide6.QtCore import QUrl

class MainWindow(QMainWindow, Ui_MainWindow):
    def __init__(self, parent=None):
        super(MainWindow, self).__init__(parent)
        self.setupUi(self)

        self.player = QMediaPlayer()
        self.audio = QAudioOutput()
        self.audio.setVolume(1)
        self.player.setAudioOutput(self.audio)

        self.pushButton_2.clicked.connect(self.voice1)

    def voice(self,num):
        a = ""
        match num:
            case 0:
                a = "http://dict.youdao.com/dictvoice?type=1&audio="
            case 1:
                a = "http://dict.youdao.com/dictvoice?type=0&audio="
        self.player.setSource(QUrl(a+self.lineEdit.text()))
        self.player.play()
        self.thread.terminate()

   def voice1(self):
        self.thread = Threads.Thread(t=0)
        self.thread.finishSignal.connect(self.voice)
        self.thread.start()
```

And in `Threads.py`:
```python
from PySide6.QtCore import QThread, Signal

class Thread(QThread):
    finishSignal = Signal(int)

    def __init__(self, t, parent=None):
        super(msgThread, self).__init__(parent)
        self.t = t

    def run(self):
        self.finishSignal.emit(int(self.t)) 
```

**Comment by erictraut** [comment]
This is not a self-contained, minimal code snippet. It references a file called `ui_main` and symbols `msgThread`, `QMediaPlayer`, `QAudioOutput`, `Ui_MainWindow`. Without understanding how these symbols are defined, we won't be able to repro the problem you're seeing. Please take another shot at producing a self-contained, minimal code sample that exhibits the problem.

It looks like you're using the library `PySide6`. Which version of the library do you have installed? Have you tried upgrading to the latest version?

**Comment by Pickaxe233** [comment]
My PySide6 is the latest, and I have the same problem in PyQt5.
And here's [ui_main.py.txt](https://github.com/microsoft/pylance-release/files/10683691/ui_main.py.txt), it's converted from qt .ui file by an extension called [Qt for Python](https://marketplace.visualstudio.com/items?itemName=seanwu.vscode-qt-for-python). `Ui_MainWindow` is included in it.
The `msgThread` is same as the `Thread` up there, only differs the class name.
For `QMediaPlayer``QAudioOutput`, I don't know how to take shot. They are a part of PySide6.

**Comment by Pickaxe233** [comment]
My PySide6 is the latest, and I have the same problem in PyQt5.
And here's [ui_main.py.txt](https://github.com/microsoft/pylance-release/files/10683691/ui_main.py.txt), it's converted from qt .ui file by an extension called [Qt for Python](https://marketplace.visualstudio.com/items?itemName=seanwu.vscode-qt-for-python). `Ui_MainWindow` is included in it.
The `msgThread` is same as the `Thread` up there, only differs the class name.
For `QMediaPlayer` `QAudioOutput`, I 'm sorry that Idon't know how to take shot. They are a part of PySide6.

**Comment by MrTribbs** [comment]
I have the same error (in a sense) with PySide6 6.4.2 (made sure its upgraded) I'm using VSC.
The clicked in `button.clicked.connect(self.the_button_was_clicked)` gives the warning "Cannot access member "clicked" for type "QPushButton" Member "clicked" is unknown"

Which the code still functions when ran, I just do not like seeing errors where there should't be one.

```python
import sys
from PySide6.QtCore import QSize
from PySide6.QtWidgets import QApplication, QMainWindow, QPushButton

class MainWindow(QMainWindow):
    def __init__(self):
        super().__init__()
        
        self.setWindowTitle("My App")
        
        self.setFixedSize(QSize(400, 300))
        
        button = QPushButton("Press Me!")
        button.setCheckable(True)
        button.clicked.connect(self.the_button_was_clicked)
        
        self.setCentralWidget(button)
        
    def the_button_was_clicked(self):
        print("Clicked!")
        
app = QApplication(sys.argv)

window = MainWindow()
window.show()

app.exec_()
```

**Comment by erictraut** [comment]
This looks to be a bug in the PySide6 package. It ships with integrated type stub files, but the type stub definitions don't include any definition for an attribute called `clicked`. Please report the bug to the maintainers of PySide6.

**Comment by Pickaxe233** [comment]
> This looks to be a bug in the PySide6 package. It ships with integrated type stub files, but the type stub definitions don't include any definition for an attribute called `clicked`. Please report the bug to the maintainers of PySide6.

OK, I'll report it. Sorry for bothering you.

**Comment by golgor** [comment]
For anyone finding this Issue, this problem seems to be fixed with version 6.5.0 of PySide6 that was released 4th of April 2023.
